tenable_io/client.py

```diff
@@ -46,20 +43,35 @@ def __init__(
         self._endpoint = endpoint
         self._impersonate = impersonate
 
-        self._headers = {
-            u'X-ApiKeys': u'accessKey=%s; secretKey=%s;' % (self._access_key, self._secret_key),
-            u'User-Agent': u'TenableIOSDK Python/%s' % ('.'.join([str(i) for i in sys.version_info][0:3]))
-        }
-
-        if impersonate is not None:
-            self._headers[u'X-Impersonate'] = u'username=%s' % self._impersonate
-
+        self._init_session()
         self._ini_api()
         self._init_helpers()
 
+    def _init_session(self):
+        """
+        Initializes the requests session
+        """
+        retries = Retry(
+                    total=3,
+                    status_forcelist={429}, 
+                    backoff_factor=0.1, 
+                    respect_retry_after_header=True
+        )
+        adapter = requests.adapters.HTTPAdapter(retries)
+        self._session = requests.Session()
+        self._session.mount('https://', adapter)
```

Comment:
If I recall correctly someone internal did in fact utilize the configurable service endpoint feature (`endpoint` in `tenable_io.ini`), however, I do not know if it was only a different hostname or different protocol all together. To answer your question, the value lies on that the SDK does not make assumption whether the API is available on HTTP or HTTPS. Perhaps it's an internal team testing against a test API instance, testing against a mock service, an outbound proxy for who-knows-what reasons... etc. I don't know and therefore I am not making an assumption. What protocol the service is available on is IMHO not an SDK-concern.

I however do not know the Tenable.io product as well as you do, so if you know for a fact that API will never be available on anything other than HTTPS. I suggest we make it impossible to configure an endpoint with HTTP to eliminate any chance an user doing so and be stuck on why the SDK not functioning correctly.